sunpy/physics/limb_darkening.py

```diff
@@ -0,0 +1,82 @@
+import copy as c
+import sunpy.map
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+
+__author__ = ["Norbert Gyenge"]
+
+
+def limbdark(observation, limb_cut=True):
+    """
+    Calculates the limb darkening function for a specified wavelength
+    and removes any limb darkening effect from the input image.
+    A 5th order polynomial is fit to the limb darkening constants obtained from
+    Astrophysical Quantities by Allan. Only in the wavelength
+    range between 4000 and 15000 A.
+    Parameters
+    ----------
+        observation - SunPy Map object
+            SDO observation
+        limb_cut - Default: True
+            Background (r>1) values will be repleaced by np.nan
+    Returns
+    -------
+        observation - Corrected sunpy map object
+        The original map will be kept.
+    Notes
+    -----
+        IDL code equavalent:
+        http://hesperia.gsfc.nasa.gov/ssw/packages/nrl/idl/nrlgen/analysis/limb_dark.pro
+    Reference
+    ---------
+        Cox, Arthur N., Allen's astrophysical quantities, 4th ed. Publisher:
+            New York: AIP Press; Springer, 2000. ISBN: 038798746
+    Examples
+    --------
+        >>> import sunpy.map
+        >>> mymap = sunpy.map.Map('test.fits')
+        >>> newmap = limb_darkening(mymap, limb_cut=True)
+        >>> mymap.peek()
+        >>> newmap.peek()"""
+
+    if observation.wavelength.value < 4000 or observation.wavelength.value > 15000:
+        raise ValueError("The wavelength of the observation must be between 4000 and 15000 A.")
+
+    observation = c.copy(observation)
```

Comment:
This should be not be here, have a look at how other methods inside map do it: i.e. https://github.com/sunpy/sunpy/blob/master/sunpy/map/mapbase.py#L1324

You should construct a new meta dict and a new data array as needed and then call the `_new_instance` method to return a new map rather than copying the original as that might lead to unexpected behavior.